### PR TITLE
Gh 328 add from hdfs

### PR DESCRIPTION
**Pull request by p013570**

### CODE REVIEW

**Comment by p013570** [comment]
This is the main fail to focus on: [AddElementsFromHdfsHandler](https://github.com/gchq/Gaffer/blob/gh-328-add-from-hdfs/store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/hdfs/handler/AddElementsFromHdfsHandler.java)

**Comment by p013570** [comment]
See the migration steps for the issue https://github.com/gchq/Gaffer/issues/328 for a list of the significant changes.

File: library/hdfs-library/src/main/java/uk/gov/gchq/gaffer/hdfs/operation/mapper/GafferMapper.java [hunk 1]
```diff
@@ -20,16 +20,18 @@
...
 import java.io.IOException;
 import java.io.UnsupportedEncodingException;
 
+import static uk.gov.gchq.gaffer.hdfs.operation.handler.job.factory.AddElementsFromHdfsJobFactory.MAPPER_GENERATOR;
+import static uk.gov.gchq.gaffer.hdfs.operation.handler.job.factory.AddElementsFromHdfsJobFactory.SCHEMA;
+import static uk.gov.gchq.gaffer.hdfs.operation.handler.job.factory.AddElementsFromHdfsJobFactory.VALIDATE;
 
 /**
- * An <code>AbstractAddElementsFromHdfsMapper</code> is a {@link Mapper} that uses a
+ * An <code>GafferMapper</code> is a {@link Mapper} that uses a
```
Comment (gaffer01) [inline library/hdfs-library/src/main/java/uk/gov/gchq/gaffer/hdfs/operation/mapper/GafferMapper.java]: Minor - Should be "A" not "An"

File: store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/hdfs/handler/AddElementsFromHdfsHandler.java [hunk 2]
```diff
@@ -52,9 +79,98 @@ public void doOperation(final AddElementsFromHdfs operation, final AccumuloStore
...
+
+        // Sample data for split points and split the table
+        final Class<? extends MapperGenerator> mapperGeneratorClass;
+        try {
+            mapperGeneratorClass = Class.forName(operation.getMapperGeneratorClassName()).asSubclass(MapperGenerator.class);
+        } catch (final ClassNotFoundException e) {
+            throw new IllegalArgumentException("Mapper generator class name was invalid: " + operation.getMapperGeneratorClassName(), e);
+        }
+
+        final String outputPath = "/tmp/" + context.getJobId() + "/output";
```
Comment (gaffer01) [inline store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/hdfs/handler/AddElementsFromHdfsHandler.java]: Should this be taken from an option on the operation? Putting split points from a table of data which has visibilities to protect the data into /tmp could be unsecure.

**Comment by p013570** [comment]
@gaffer01 - thanks, good spot. I've created a workingPath field on AddElementsFromHdfs and we will use that instead of the /tmp.

**Comment by p013570** [comment]
I think we should add a "numSplits" field to SplitStore as it might be useful for other store implementations.

Also, there are some edge cases when you might not want Accumulo to Sample and Split before running the AddElementsFromHdfs, so we will need to provide an option to disable that.

Closing while we implement the above.

**Pull request merged**